nispor: Use ethtool link mode for speed/auto-negotiation/duplex.

cathay4t opened the pull request:
Using ethtool link mode information from nispor to replace the old hack on
ioctl codes.

cathay4t commented:
Waiting PR #1592

ffmancera commented on libnmstate/ifaces/ethernet.py:
```diff
@@ -79,6 +78,20 @@ def is_sriov(self):
+        )
+
+    @property
+    def duplex(self):
```
Please, could you add unit tests covering the new properties? Thanks!

cathay4t replied:
Updated existing unit test case to cover these new properties.

ffmancera reviewed (changes requested):
Just a nit. Thank you!

cathay4t committed:
nispor: Use ethtool link mode for speed/auto-negotiation/duplex.

Using ethtool link mode information from nispor to replace the old
hack on ioctl codes.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
